redash/query_runner/clickhouse.py

```diff
@@ -0,0 +1,120 @@
+import json
+import logging
+from redash.query_runner import *
+from redash.utils import JSONEncoder
+logger = logging.getLogger(__name__)
+
+try:
+    import requests
+    enabled = True
+except ImportError as e:
+    logger.info(str(e))
+    enabled = False
+
+
+class ClickHouse(BaseSQLQueryRunner):
+    noop_query = "SELECT 1"
+
+    @classmethod
+    def configuration_schema(cls):
+        return {
+            "type": "object",
+            "properties": {
+                "user": {
+                    "type": "string",
+                    "default": "default"
+                },
+                "password": {
+                    "type": "string"
+                },
+                "host": {
+                    "type": "string",
+                    "default": "127.0.0.1"
+                },
+                "port": {
+                    "type": "number",
+                    "default": 8123
+                },
+                "dbname": {
+                    "type": "string",
+                    "title": "Database Name"
+                }
+            },
+            "required": ["dbname"],
+            "secret": ["password"]
+        }
+
+    @classmethod
+    def type(cls):
+        return "clickhouse"
+
+    def __init__(self, configuration):
+        super(ClickHouse, self).__init__(configuration)
+
+    def _get_tables(self, schema):
+        query = "SELECT database, table, name FROM system.columns WHERE database NOT IN ('system')"
+
+        results, error = self.run_query(query, None)
+
+        if error is not None:
+            raise Exception("Failed getting schema.")
+
+        results = json.loads(results)
+
+        for row in results['rows']:
+            table_name = '{}.{}'.format(row['database'], row['table'])
+
+            if table_name not in schema:
+                schema[table_name] = {'name': table_name, 'columns': []}
+
+            schema[table_name]['columns'].append(row['name'])
+
+        return schema.values()
+
+    def _send_query(self, data, stream=False):
+        url = 'http://{host}:{port}'.format(host=self.configuration['host'], port=self.configuration['port'])
```

Comment:
http and not https? why not let the user specify the full url instead of host/port pair?